### PR TITLE
Remove unnecessary if statements

### PR DESCRIPTION
**Pull request by BBosman**
Similar to #950.

All these `if`-checks have nearly identical checks just before, making these unnecessary as the result won't be different.

### CODE REVIEW

File: src/OrcasEngine/Engine/Expander.cs [hunk 1]
```diff
@@ -1394,10 +1394,7 @@ private static Type GetTypeForStaticMethod(string typeName)
                 if (Environment.GetEnvironmentVariable("MSBUILDENABLEALLPROPERTYFUNCTIONS") == "1")
                 {
                     // We didn't find the type, so go probing. First in System
-                    if (objectType == null)
-                    {
-                        objectType = GetTypeFromAssembly(typeName, "System");
-                    }
+                    objectType = GetTypeFromAssembly(typeName, "System");
```
Comment (cdmihai) [inline src/OrcasEngine/Engine/Expander.cs]: see https://github.com/Microsoft/msbuild/pull/963/files#r76450977

**Comment by cdmihai** [comment]
Looks good, except for the change in `src/OrcasEngine`

**Comment by BBosman** [comment]
I've updated the PR to remove the `src/OrcasEngine` change from it.

**Comment by rainersigwald** [comment]
LGTM, thanks.

**Pull request merged**